src/inc/eventtracebase.h

```diff
@@ -109,7 +109,7 @@ enum EtwThreadFlags
 #define ETWFireEvent(EventName)
 
 #define ETW_TRACING_INITIALIZED(RegHandle) (TRUE)
-#define ETW_EVENT_ENABLED(Context, EventDescriptor) (EventPipeHelper::Enabled() || XplatEventLogger::IsEventLoggingEnabled())
+#define ETW_EVENT_ENABLED(Context, EventDescriptor) (EventPipeHelper::Enabled() || EventEnabled##EventDescriptor())
 #define ETW_CATEGORY_ENABLED(Context, Level, Keyword) (EventPipeHelper::Enabled() || XplatEventLogger::IsEventLoggingEnabled())
```

Comment:
I'm thinking for ETW_CATEGORY_ENABLED and ETW_TRACING_CATEGORY_ENABLED, most of the calls can probably be replaced with ETW_EVENT_ENABLED pretty easily.  There will be some where we'd need to pick an event to be representative of the configuration, but I think we can do that.  Then, we could remove the two defines, and with them the implementation for XplatEventLogger::IsEventLogEnabled() and the environment variable.  Do you want to take a crack at this?

I'd love to get rid of COMPlus_EventLogEnabled once and for all, and I like the idea of not having some paths that call it and some paths that don't.

Reply:
👍 Agree. Perhaps in a separate PR? See my comment here: https://github.com/dotnet/coreclr/issues/14327#issuecomment-339592313.

Reply:
I think that's fine.  Thanks!